silkworm/rpc/core/evm_trace.hpp

```diff
@@ -144,15 +153,21 @@ class VmTraceTracer : public silkworm::EvmTracer {
     void on_precompiled_run(const evmc_result& result, int64_t gas, const silkworm::IntraBlockState& intra_block_state) noexcept override;
 
   private:
-    bool is_precompile_{false};
+    void fill_call_gas_info(TraceOp& trace_op,
+                            const evmone::ExecutionState& execution_state,
+                            const intx::uint256* stack_top,
+                            int stack_height,
+                            const silkworm::IntraBlockState& intra_block_state);
+
     VmTrace& vm_trace_;
     std::int32_t transaction_index_;
     std::stack<std::string> index_prefix_;
     std::stack<std::reference_wrapper<VmTrace>> traces_stack_;
     const char* const* opcode_names_ = nullptr;
+    const evmc_instruction_metrics* metrics_ = nullptr;
     std::stack<int64_t> start_gas_;
     std::stack<TraceMemory> trace_memory_stack_;
-    const evmc_instruction_metrics* metrics_ = nullptr;
+    std::unique_ptr<FixCallGasInfo> fix_call_gas_info_;
```

Comment:
Let's use `std::optional` instead of `std::unique_ptr` here, as outlined in #2192 for `DebugTracer`.

Reply:
Done